test_runner/src/test/kotlin/ftl/reports/xml/JUnitXmlTest.kt

```diff
@@ -437,4 +440,32 @@ junit.framework.Assert.fail(Assert.java:50)</failure>
         """.trimIndent()
         assertThat(merged).isEqualTo(expected)
     }
+    @Test
+    fun `parse ftl quirks`() {
+        val crashingAllSuitesMessage = """
+            <?xml version='1.0' encoding='UTF-8' ?>
+            <testsuites>
+              <testsuite name="EarlGreyExampleSwiftTests" tests="3" failures="0" errors="0" skipped="0" time="10.0" hostname="localhost">
+                <testcase name="a()" classname="a" time="1.0">
+                <failure> java.net.ConnectException: Failed to connect to ... at &#8;&#8;&#8;(Coroutine boundary.&#8;(&#8;)</failure>
+                </testcase>
+                <testcase name="b()" classname="b" time="2.0"/>
+                <testcase name="c()" classname="c" time="7.0"/>
+              </testsuite>
+            </testsuites>
+        """.trimIndent()
+        val crashingOneSuiteMessage = """
+            <?xml version='1.0' encoding='UTF-8' ?>
+              <testsuite name="EarlGreyExampleSwiftTests" tests="3" failures="0" errors="0" skipped="0" time="10.0" hostname="localhost">
+                <testcase name="a()" classname="a" time="1.0">
+                <failure> java.net.ConnectException: Failed to connect to ... at &#8;&#8;&#8;(Coroutine boundary.&#8;(&#8;)</failure>
+                </testcase>
+                <testcase name="b()" classname="b" time="2.0"/>
+                <testcase name="c()" classname="c" time="7.0"/>
+              </testsuite>
+        """.trimIndent()
+
+        parseAllSuitesXml(crashingAllSuitesMessage)
+        parseOneSuiteXml(crashingOneSuiteMessage)
```

Comment:
should we also assert that the parsed value matches the expected value? Currently this test will pass as long as parsing doesn't fail. We probably want to assert ` <failure>...` is decoded correctly as well.

Reply:
You have right, Im add asserts and split tests to two. One for one suite and one for all suites